pkg/roachprod/cloud/cluster_cloud.go

```diff
@@ -346,14 +355,44 @@ func CreateCluster(l *logger.Logger, opts []*ClusterCreateOpts) error {
 			p = (p + 1) % providerCount
 		}
 
+		var vms vm.List
+		var vmsLock syncutil.Mutex
+		// Create VMs in parallel across all providers.
+		// Each provider will return the list of VMs it created, and we append
+		// them to the cached Cluster.
 		if err := vm.ProvidersParallel(o.CreateOpts.VMProviders, func(p vm.Provider) error {
-			return p.Create(l, vmLocations[p.Name()], o.CreateOpts, o.ProviderOptsContainer[p.Name()])
+			vmList, err := p.Create(l, vmLocations[p.Name()], o.CreateOpts, o.ProviderOptsContainer[p.Name()])
+			if err != nil {
+				return err
+			}
+			vmsLock.Lock()
+			defer vmsLock.Unlock()
+			vms = append(vms, vmList...)
+			return nil
 		}); err != nil {
-			return err
+			return c, err
 		}
+
+		c.VMs = append(c.VMs, vms...)
 	}
 
-	return nil
+	// As VMs are created in parallel, sort them to ensure consistent ordering.
+	// SyncedCluster.Wait() tends to be flaky if the VMs are not sorted and VMs
+	// become available in an unexpected order.
+	// Sort is also used in ListCloud() to ensure consistent ordering.
+	sort.Sort(c.VMs)
+
+	// If VMs were created, set the cluster user to the user of the first VM.
```

Comment:
If no VMs were created, this would indicate an error, right? (Since every `Create` must have returned `nil, err`.)
Thus, the check below may not be enough, since we want to return `nil, err` in this case.